src/neuroconv/datainterfaces/ecephys/spikeglx/spikeglxconverter.py

```diff
@@ -0,0 +1,107 @@
+"""The simplest, easiest to use class for converting all SpikeGLX data in a folder."""
+from pathlib import Path
+from typing import List, Optional
+
+import numpy as np
+from pynwb import NWBFile
+
+from .spikeglxdatainterface import SpikeGLXRecordingInterface
+from .spikeglxnidqinterface import SpikeGLXNIDQInterface
+from ....nwbconverter import ConverterPipe
+from ....tools.nwb_helpers import make_or_load_nwbfile
+from ....utils import FolderPathType, get_schema_from_method_signature
+
+
+class SpikeGLXConverterPipe(ConverterPipe):
+    """Primary conversion class for handling multiple SpikeGLX data streams."""
+
+    @classmethod
+    def get_source_schema(cls):
+        source_schema = get_schema_from_method_signature(method=cls.__init__, exclude=["streams"])
```

Comment:
oh, because you need the folder path to get the streams. Is there any way we can provide a user with the option to select specific streams?

Reply:
See the function `get_streams` just below this one.

I suppose it could be included in the source schema as a list of strings, but what I was really imagining for this on the GUIDE side (and in general) is it really actually ought to be a `Literal` of the number of actual choices for streams (so a dropdown/enum rather than freeform)

Reply:
I suppose in jsonschema this would translate to:

```json
{
  "type": "array",
    "items": {
      "enum" : streams
  },
  "uniqueItems": true
}
```

I'm not sure it would be easy to express the uniqueness in the function signature, so we might need to construct that part of the jsonschema manually